python/lsst/daf/butler/tests/_testRepo.py

```diff
@@ -183,7 +184,7 @@ def _makeRecords(dataIds: Mapping[str, Iterable], universe: DimensionUniverse) -
             # Do not recurse, to keep the user from having to provide
             # irrelevant dimensions
             for other in dimension.implied:
-                if other != dimension and other.name in expandedIds and other.viewOf is None:
+                if other.name in expandedIds:
```

Comment:
If you're doing what I think you're doing, then the comment is no longer correct.

Reply:
Part of the reason `_testRepo.py` exists in the first place is to allow `PipelineTask` developers to set up a test environment without any deep knowledge of how Middleware works, including the sometimes unintuitive relationships between dimensions and the special treatment of `band`.

Would it be possible to generate a value (e.g., `band = physical_filter`) analogously to what `_fillAllKeys` does, instead of forcing the user to figure out why their test setup is giving cryptic errors both when they try to create a band and when they don't?

Reply:
Do similar changes need to be made to `addDataIdValue`? It's a bit surprising that all the code changes are in the old system.

Reply:
I agree the comment is incorrect, and I'll at least fix that (if I can't make it remain correct).

I think there may indeed be an opportunity to generate values for the missing dimensions automatically, and avoid the other branches for this ticket, and I just didn't understand where to put it.  My guess would be to have `_makeRecords` start by ensuring that its input mapping has keys for all dimensions implied by any of the dimensions it is given, using that `_fillAllKeys`-like logic.  I'll attempt that later today, but if you can confirm or reject that as a reasonable approach first I'd appreciate it.

> Do similar changes need to be made to `addDataIdValue`? It's a bit surprising that all the code changes are in the old system.

If I can fix the other code path as you've requested, then I think that means I should fix this one, too.  I suspect the reason I didn't have to was because it already made it possible (but after the NOT NULL constraint lands, necessary) to pass implied values by including them in the data ID key, so I just went around making changes in the tests like:
```
        -addDataIdValue(cls.creatorButler, "physical_filter", "l2019", instrument="dummyCam")
        +addDataIdValue(cls.creatorButler, "physical_filter", "l2019", band="l", instrument="dummyCam")
```
And I think the right place for that fix is in `_matchAnyDataId`, which could create (with `_findAllKeys`-like logic) and insert a new dimension record when there isn't one, instead of raising.  Once again, confirmation that that approach makes sense would be appreciated.

Reply:
>  My guess would be to have `_makeRecords` start by ensuring that its input mapping has keys for all dimensions implied by any of the dimensions it is given, using that `_fillAllKeys`-like logic. I'll attempt that later today, but if you can confirm or reject that as a reasonable approach first I'd appreciate it.

Yes, I think so. Unfortunately, I haven't looked at this code since last year, and I wrote it with a very vague idea of how the dimensions system works, so I can't offer any special insights.

> And I think the right place for that fix is in `_matchAnyDataId`, which could create (with `_findAllKeys`-like logic) and insert a new dimension record when there isn't one, instead of raising. Once again, confirmation that that approach makes sense would be appreciated.

I'm a little nervous about having `addDataIdValue` create dimension records for other dimensions (looks like it would be `physical_filter` in practice), but I think it's the least of the available evils.